e2e/backend_service_update_test.go

```diff
@@ -226,3 +244,76 @@ func TestBackendServiceUpdate(t *testing.T) {
 	wantBS = &compute.BackendService{LoadBalancingScheme: "INTERNAL_MANAGED"}
 	checkBackendService(t, ctx, theCloud, bs1ID, wantBS, compareLBScheme)
 }
+
+func TestBackendServiceMetadataUpdate(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	graphBuilder := rgraph.NewBuilder()
+	resUniqueIdPart := "test-bs-metadata-update"
+
+	/* Create BackendService */
+	bsName := resUniqueIdPart + "-bs"
+	wantedMetadata := map[string]string{"foo": "1", "bar": "2", "baz": "2"}
+	bsID, err := buildBackendServiceWithMetadata(graphBuilder, bsName, wantedMetadata)
+
+	expectedActions := []exec.ActionMetadata{
+		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, bsID)},
+	}
+
+	processGraphAndExpectActions(t, graphBuilder, expectedActions)
+	t.Cleanup(func() {
+		err = theCloud.BackendServices().Delete(ctx, bsID.Key)
+		if err != nil {
+			t.Logf("theCloud.BackendServices().Delete(_, %s): %v", bsID.Key, err)
+		}
+	})
+
+	/* Verify BackendService was created correctly */
+	bs, err := theCloud.BackendServices().Get(ctx, bsID.Key)
+	if err != nil {
+		t.Fatalf("theCloud.BackendServices().Get(_, %s) = (nil, %v) wanted (_, nil)", bsID.Key, err)
+	}
+
+	if df := cmp.Diff(bs.Metadatas, wantedMetadata); df != "" {
+		t.Fatalf("Metadata are not set correctly: %s", df)
+	}
+
+	/* Edit BackendService metadatas field */
+	bsID, err = buildBackendServiceWithMetadata(graphBuilder, bsName, map[string]string{"foo": "1", "bar": "2", "baz": "3"})
+	expectedActions = []exec.ActionMetadata{
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, bsID)},
+	}
+
+	processGraphAndExpectActions(t, graphBuilder, expectedActions)
+	wantedMetadata = map[string]string{"foo": "1", "bar": "2", "baz": "3"}
+
+	/* Verify Metadatas field was updated correctly */
+	bs, err = theCloud.BackendServices().Get(ctx, bsID.Key)
+	if err != nil {
+		t.Fatalf("theCloud.BackendServices().Get(_, %s) = (nil, %v) wanted (_, nil)", bsID.Key, err)
+	}
+
+	if df := cmp.Diff(bs.Metadatas, wantedMetadata); df != "" {
+		t.Fatalf("Metadata keys are not updated correctly(-got, +want): %s", df)
```

Comment:
please unify error messaged, the one in line 279 is different

Reply:
The message in line 279 is a different message because it's about setting metadatas for the first time not updating them (even though technically it might be performed the same way I did it on purpose)

Reply:
In line 279 (-got, + want) is missing, please at it

Reply:
Oh, I see. Done. Thanks.